### DISCUSSION TITLE
DetachedInstanceError on Handling ResponseValidationError with SQLAlchemy

### DISCUSSION THREAD

**Discussion by FeehGb** [fastapi/fastapi#11081] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

The error occurs when making a request to a model that retrieves clusters, specifically through the `GrupoSkuModel`. The issue is reproduced by accessing the `clusters` attribute, which should be correctly typed with `Mapped` but triggers a `DetachedInstanceError` due to the SQLAlchemy relationship handling.

Here is a simplified code snippet to demonstrate the process and how the error can be reproduced:
```python

from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import SmallInteger, String, Boolean, select, expression
from sqlalchemy.orm import declarative_base, Session
from sqlalchemy.ext.hybrid import hybrid_property

Base = declarative_base()

class GrupoSkuClusterModel(Base):
    # Placeholder for the actual model definition, assuming a back_populates to "grupo_sku"
    ...

class GrupoSkuModel(Base):
    __tablename__ = "tbl_grupo_sku"

    id: Mapped[int] = mapped_column(SmallInteger, primary_key=True)
    descricao: Mapped[str] = mapped_column(String(100), nullable=False, index=True, unique=True)
    # The error occurs because its not Mapped, should be like this -> Mapped[list["GrupoSkuClusterModel"]]
    clusters: GrupoSkuClusterModel = relationship(
        "GrupoSkuClusterModel",
        back_populates="grupo_sku",
        lazy="joined",
        cascade="all, delete-orphan"
    )
    # Additional attributes and relationships are omitted for brevity

# Example code to trigger the error
session = Session()
grupo_sku = session.query(GrupoSkuModel).first()  # Simulates retrieving a GrupoSkuModel instance
print(grupo_sku.clusters)  # Accessing the clusters attribute, which reproduces the DetachedInstanceError
```

### Description

When using FastAPI in conjunction with SQLAlchemy (without using `Mapped`), I am encountering an issue where a `ResponseValidationError` is correctly raised, but when attempting to format the error message for display (or logging), a `DetachedInstanceError` from SQLAlchemy is thrown.

### Expected Behavior

The expectation was that the `ResponseValidationError` could be handled, and its error message formatted without causing a `DetachedInstanceError` from SQLAlchemy.

### Actual Behavior

When the `ResponseValidationError` exception attempts to format its error message for display, specifically on the line `message += f"  {err}\n"`, an `sqlalchemy.orm.exc.DetachedInstanceError` exception is thrown. The error message indicates that the model instance is not bound to a session, which prevents the attribute refresh operation.

### Class that raised Exception

```python
class ResponseValidationError(ValidationException):
    def __init__(self, errors: Sequence[Any], *, body: Any = None) -> None:
        super().__init__(errors)
        self.body = body

    def __str__(self) -> str:
        message = f"{len(self._errors)} validation errors:\n"
        for err in self._errors:
            message += f"  {err}\n"
        return message
```

### Operating System

Windows

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.11

### Environment

- FastAPI Version: 0.109.0
- SQLAlchemy Version: 2.0.25
- Python Version: 3.11
- Pydantic Version: 2.5.3

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
